### PR TITLE
CP-44563: Compress tracing files with zstd

### PR DESCRIPTION
**Pull request by snwoods**

### CODE REVIEW

File: ocaml/libs/tracing/tracing.mli [hunk 1]
```diff
@@ -129,6 +129,8 @@ module Export : sig
       val set_max_file_size : int -> unit
 
       val set_trace_log_dir : string -> unit
+
+      val set_compress_tracing_files : bool -> unit
```
Comment (lindig) [inline ocaml/libs/tracing/tracing.mli:133]: It's not possible right now to observe the status of this flag, only to set it. Maybe you don't need it right now.

File: ocaml/libs/xapi-compression/xapi_compression.ml [hunk 2]
```diff
@@ -20,6 +20,12 @@ module type COMPRESSOR = sig
 
   (* Experimental decompressor which is fed from an fd and writes to a pipe *)
   val decompress_passive : Unix.file_descr -> (Unix.file_descr -> 'a) -> 'a
+
+  val compress_file :
```
Comment (lindig) [inline ocaml/libs/xapi-compression/xapi_compression.ml:24]: We can expect this to raise an exception if the file can't be found or written. Would be good to document it.

File: ocaml/libs/xapi-compression/xapi_compression.mli [hunk 3]
```diff
@@ -20,6 +20,12 @@ module type COMPRESSOR = sig
 
   (* Experimental decompressor which is fed from an fd and writes to a pipe *)
   val decompress_passive : Unix.file_descr -> (Unix.file_descr -> 'a) -> 'a
+
+  val compress_file :
```
Comment (lindig) [inline ocaml/libs/xapi-compression/xapi_compression.mli:24]: This needs documentation.

**Review by lindig — APPROVED** [review]

**Review by psafont — APPROVED** [review]

File: ocaml/libs/xapi-compression/xapi_compression.mli [hunk 4]
```diff
@@ -20,6 +20,14 @@ module type COMPRESSOR = sig
...
+       (Unix.file_descr -> (Unix.file_descr -> unit) -> 'a)
+    -> file_path:string
+    -> file_ext:string
+    -> unit
+  (* Uses the compress function given to compress the specified file.
```
Comment (psafont) [inline ocaml/libs/xapi-compression/xapi_compression.mli:29]: Documentation comments start with `(**`

**Pull request merged**